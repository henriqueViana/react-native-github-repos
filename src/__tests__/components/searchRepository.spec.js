import React from 'react'
import { shallow } from 'enzyme'
import { TextInput } from 'react-native'
import { FontAwesome } from '@expo/vector-icons'

import SearchRepository from '../../components/searchRepository'

describe('-----SearchRepository-----', () => {

  it('should renders correcty SearchRepository', () => {
    const wrapper = shallow(<SearchRepository />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should render a TextInput child component', () => {
    const wrapper = shallow(<SearchRepository />)

    const childComponent = wrapper.find(TextInput)
    expect(childComponent).toHaveLength(1)
  })

  it('should render a FontAwesome icon child component', () => {
    const wrapper = shallow(<SearchRepository />)

    const childComponent = wrapper.find(FontAwesome)
    expect(childComponent).toHaveLength(1)
  })  
})