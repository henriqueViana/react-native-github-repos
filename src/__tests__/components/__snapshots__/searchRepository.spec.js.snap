// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`-----SearchRepository----- should renders correcty SearchRepository 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "#d2e0e2",
      "marginBottom": 70,
      "padding": 5,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFF",
        "flexDirection": "row",
        "paddingHorizontal": 10,
      }
    }
  >
    <Icon
      allowFontScaling={false}
      color="#000"
      name="search"
      size={15}
    />
    <TextInput
      allowFontScaling={true}
      placeholder="Nome do usuÃ¡rio"
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "#FFF",
          "flex": 1,
          "padding": 10,
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
</View>
`;
